apps/pkeyutl.c

```diff
@@ -371,14 +412,24 @@ int pkeyutl_main(int argc, char **argv)
         buf_outlen = kdflen;
         rv = 1;
     } else {
-        rv = do_keyop(ctx, pkey_op, NULL, (size_t *)&buf_outlen,
-                      buf_in, (size_t)buf_inlen);
+        if (rawin)
+            rv = do_raw_keyop(pkey_op, ctx, md, pkey, in, NULL, 0,
+                              NULL, (size_t *)&buf_outlen, &saved_mctx);
```

Comment:
Would it be more graceful to not use `saved_mctx`, and let `do_raw_keyop` do the `app_malloc` call for `buf_out` instead? Then the `if (rv > 0 && buf_outlen != 0)` code could be moved into the "else" below, i.e. so it only gets called on the do_keyop branch.

Reply:
Right, this is more graceful, fixed.